src/screens/ScreenNewsList.js

```diff
@@ -0,0 +1,144 @@
+import React, { Component } from 'react';
+import { View, StyleSheet } from 'react-native';
+import { fetchNews } from '../api';
+import { NewsListHeader, NewsListBody } from '../components';
+
+export default class ScreenNewsList extends Component {
+  constructor(props) {
+    super(props);
+    const { articles: art } = props;
+
+    this.state = {
+      isRefreshing: false,
+      articles: art,
+      page: 0,
+      hasArticles: art.length > 0,
+    };
+
+    this.refresh = this.refresh.bind(this);
+    this.startLoading = this.startLoading.bind(this);
+    this.stopLoading = this.stopLoading.bind(this);
+    this.handleCategoriesPress = this.handleCategoriesPress.bind(this);
+    this.handleFavoritesPress = this.handleFavoritesPress.bind(this);
+    this.handleNewsPressed = this.handleNewsPressed.bind(this);
+    this.loadMore = this.loadMore.bind(this);
+  }
+
+  componentDidMount() {
+    this.refresh();
+  }
+
+  async refresh() {
+    const { isRefreshing } = this.state;
+    if (isRefreshing) {
+      console.log('prevented refreshing');
+      return;
+    }
+    const { hasArticles } = this.state;
+    if (hasArticles) {
+      this.setState({ hasArticles: false });
+      return;
+    }
+
+    this.startLoading();
+    console.log('refreshing');
+
+    const page = 0;
+
+    try {
+      const articles = await fetchNews(page);
+      this.setState({ articles, page });
+    } catch (e) {
+      // intentionally empty
+    }
+
+    this.stopLoading();
+  }
+
+  async loadMore() {
+    const { isRefreshing } = this.state;
+    if (isRefreshing) {
+      console.log('prevented loading more');
+      return;
+    }
+    this.startLoading();
+
+    let { page } = this.state;
```

Comment:
const, dont' use `let`

Reply:
I used let because it lets me change the variable later. Connected to page: page+1 comment

Reply:
Avoid `let` as much as possible, 99% can be written with `const`